htdocs/core/lib/files.lib.php

```diff
@@ -2193,7 +2193,7 @@ function dol_compress_dir($inputdir, $outputfile, $mode = "zip", $excludefiles =
 					if (!$file->isDir()) {
 						// Get real and relative path for current file
 						$filePath = $file->getRealPath();
-						$relativePath = ($rootdirinzip ? $rootdirinzip.'/' : '').substr($filePath, strlen($inputdir) + 1);
+						$relativePath = ($rootdirinzip ? $rootdirinzip.'/' : '').substr($filePath, strlen(realpath($inputdir)) + 1);
```

Comment:
I just try v16 fix, but doesn't fix at all. A folder was create with a different name.

Reply:
I can't reproduce the trouble. Can you describe how to reproduce the bug ?

Reply:
Avec le fix de la v16 : lors de l'appel de la fonction dol_compress_dir 
Document à archiver :
![image](https://user-images.githubusercontent.com/85485123/178692643-3a9d7f7b-9cbb-47a4-866e-b73c959f0c55.png)

Résultat : Une dossier est créé avec le nom à moitié coupé et une extension.
![image](https://user-images.githubusercontent.com/85485123/178693049-121c4a25-6c1e-4eac-ab0d-ae14e37e1168.png)



Avec notre méthode l'archive contient seulement les fichiers avec les bons noms ( ne rajoute pas de nouveau dossier)
Cordialement

Reply:
What are the values that you pass to dol_compress_dir ?

Reply:
I don't have the same environment as my colleague, but I think the issue was with a discrepancy between `$filePath` and `$inputdir` because of a symbolic link being resolved in one of the path and not in the other.

For instance it could have been:

- $filePath = `/var/www/html/mycustomer/dolibarr/documents/test/directory/subdir/myfile.pdf`
- $inputdir = `/customers/mycustomer/dolibarr/documents/test/directory`

With `/customers/mycustomer` being a symbolic link to `/var/www/html/mycustomer`.

The current code expects the two variables to start with the same initial path, but the symbolic makes this assumption incorrect. So in our example, assuming `$rootindirzip` is empty, the value of `$relativePath` would be:
- `ry/subdir/myfile.pdf` (buggy) instead of
- `subdir/myfile.pdf` (expected)

Adding `realpath` helps because it resolves the symlink.